vm-allocator/src/system.rs

```diff
@@ -51,24 +52,25 @@ impl SystemAllocator {
     /// Creates a new `SystemAllocator` for managing addresses and irq numvers.
     /// Can return `None` if `base` + `size` overflows a u64
     ///
-    /// * `io_base` - The starting address of IO memory.
-    /// * `io_size` - The size of IO memory.
+    /// * `io_base` - (X86) The starting address of IO memory.
+    /// * `io_size` - (X86) The size of IO memory.
     /// * `mmio_base` - The starting address of MMIO memory.
     /// * `mmio_size` - The size of MMIO memory.
     /// * `mmio_hole_base` - The starting address of MMIO memory in 32-bit address space.
     /// * `mmio_hole_size` - The size of MMIO memory in 32-bit address space.
     /// * `apics` - (X86) Vector of APIC's.
     ///
     pub fn new(
-        io_base: GuestAddress,
-        io_size: GuestUsize,
+        #[cfg(target_arch = "x86_64")] io_base: GuestAddress,
+        #[cfg(target_arch = "x86_64")] io_size: GuestUsize,
```

Comment:
My original consideration to screen IO things with conditions is:
1. IO is not for ARM, having it in binary looks weird.
2. Hiding IO bus may avoid potential mistakes in code. It should be wrong if some ARM code is using IO operating/bus, but this will be impossible if IO things are not present. I also planed to make PciBarRegionType::IORegion x86_64 only when coming to support PCI.

Reply:
Okay!